src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -72,16 +78,18 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
             [this](const std::vector<char>& message)
             {
                 std::cout << "DatabaseFeedManager::update" << std::endl;
-                auto eventContext = std::make_shared<EventContext>(
-                    EventContext {.indexerConnector = m_indexerConnector, .message = message});
-                auto eventDecoder = std::make_shared<EventDecoder>();
-                eventDecoder->setLast(std::make_shared<StoreModel>());
-                eventDecoder->setLast(std::make_shared<FeedIndexer>());
-                eventDecoder->setLast(std::make_shared<ScanDispatcher>());
+                auto eventContext = std::make_shared<TemplatedEventContext<TIndexerConnector>>(
+                    TemplatedEventContext<TIndexerConnector> {.indexerConnector = m_indexerConnector,
+                                                              .message = message});
+                auto eventDecoder = std::make_shared<TemplatedEventDecoder<TIndexerConnector>>();
+                eventDecoder->setLast(std::make_shared<TemplatedStoreModel<TIndexerConnector>>());
+                eventDecoder->setLast(std::make_shared<TemplatedFeedIndexer<TIndexerConnector>>());
+                eventDecoder->setLast(std::make_shared<TemplatedScanDispatcher<TIndexerConnector>>());
```

Comment:
There we need to use the typename passed in the template because for test cases they are the trampoline type classes.
Finally, this part of the code was changed according to the commit of Octavio, which implies fewer classes to templatize.